src/ColorVectorSpace.jl

```diff
@@ -314,6 +314,10 @@ end
 (.*){T<:Number}(A::AbstractArray{T}, b::TransparentGray) = mul(A, b)
 (.*){T<:Number}(b::TransparentGray, A::AbstractArray{T}) = mul(b, A)
 
+varm{C<:AbstractGray}(v::AbstractArray{C}, s::AbstractGray; corrected::Bool=true) =
+        varm(map(gray,v),gray(s); corrected=corrected)
+real{C<:AbstractGray}(x::Type{C}) = C
```

Comment:
Oh, nevermind, as usual I was misreading. Most of the time, irrelevant variables are not named, e.g.,
```jl
real{C<:AbstractGray}(::Type{C}) = C
```
Now my concern is the following:
```jl
julia> Gray{Float32} <: Real
false
```
Maybe it would help to describe specifically what you're trying to accomplish.